sycl/test-e2e/DeviceLib/cmath_test.cpp

```diff
@@ -1,11 +1,17 @@
 // DEFINE: %{mathflags} = %if cl_options %{/clang:-fno-fast-math%} %else %{-fno-fast-math%}
 
-// UNSUPPORTED: hip
+// TODO reenable this test when -ffast-math works with cmath_test for SPIR-V
+// compilation
+// REQUIRES: cuda
 // RUN: %{build} -fno-builtin %{mathflags} -o %t.out
 // RUN: %{run} %t.out
 
 // RUN: %{build} -fno-builtin -fsycl-device-lib-jit-link %{mathflags} -o %t.out
 // RUN: %if !gpu %{ %{run} %t.out %}
+//
+// // Check that --fast-math works with cmath funcs
+// RUN: %{build} -fno-builtin %{mathflags} -o %t.out -ffast-math -DSYCL_E2E_FASTMATH
```

Comment:
Is there anything we could do here to only enable the `-ffast-math` testing for cuda? Some other tests have interesting macros like: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/Reduction/reduction_range_1d_dw.cpp#L1

The `REQUIRES` above is a bit brutal since it will disable this test for all the other targets.

Reply:
Yep agreed have changed to a conditional run for CUDA for fast math case